zarr/tests/test_convenience.py

```diff
@@ -85,3 +88,235 @@ def test_lazy_loader():
     assert_array_equal(foo, loader['foo'])
     assert_array_equal(bar, loader['bar'])
+
+
+def test_copy_store():
+
+    # no paths
+    source = dict()
+    source['foo'] = b'xxx'
+    source['bar'] = b'yyy'
+    dest = dict()
+    copy_store(source, dest)
+    assert len(dest) == 2
+    for key in source:
+        assert source[key] == dest[key]
+
+    # with source path
+    source = dict()
+    source['foo'] = b'xxx'
+    source['bar/baz'] = b'yyy'
+    source['bar/qux'] = b'zzz'
+    # paths should be normalized
+    for source_path in 'bar', 'bar/', '/bar', '/bar/':
+        dest = dict()
+        copy_store(source, dest, source_path=source_path)
+        assert len(dest) == 2
+        for key in source:
+            if key.startswith('bar/'):
+                dest_key = key.split('bar/')[1]
+                assert source[key] == dest[dest_key]
+            else:
+                assert key not in dest
+
+    # with dest path
+    source = dict()
+    source['foo'] = b'xxx'
+    source['bar/baz'] = b'yyy'
+    source['bar/qux'] = b'zzz'
+    # paths should be normalized
+    for dest_path in 'new', 'new/', '/new', '/new/':
+        dest = dict()
+        copy_store(source, dest, dest_path=dest_path)
+        assert len(dest) == 3
+        for key in source:
+            dest_key = 'new/' + key
+            assert source[key] == dest[dest_key]
+
+    # with source and dest path
+    source = dict()
+    source['foo'] = b'xxx'
+    source['bar/baz'] = b'yyy'
+    source['bar/qux'] = b'zzz'
+    # paths should be normalized
+    for source_path in 'bar', 'bar/', '/bar', '/bar/':
+        for dest_path in 'new', 'new/', '/new', '/new/':
+            dest = dict()
+            copy_store(source, dest, source_path=source_path, dest_path=dest_path)
+            assert len(dest) == 2
+            for key in source:
+                if key.startswith('bar/'):
+                    dest_key = 'new/' + key.split('bar/')[1]
+                    assert source[key] == dest[dest_key]
+                else:
+                    assert key not in dest
+                    assert ('new/' + key) not in dest
+
+    # with excludes/includes
+    source = dict()
+    source['foo'] = b'xxx'
+    source['bar/baz'] = b'yyy'
+    source['bar/qux'] = b'zzz'
+    # single excludes
+    dest = dict()
+    excludes = 'f.*'
+    copy_store(source, dest, excludes=excludes)
+    assert len(dest) == 2
+    assert 'foo' not in dest
+    # multiple excludes
+    dest = dict()
+    excludes = 'b.z', '.*x'
+    copy_store(source, dest, excludes=excludes)
+    assert len(dest) == 1
+    assert 'foo' in dest
+    assert 'bar/baz' not in dest
+    assert 'bar/qux' not in dest
+    # excludes and includes
+    dest = dict()
+    excludes = 'b.*'
+    includes = '.*x'
+    copy_store(source, dest, excludes=excludes, includes=includes)
+    assert len(dest) == 2
+    assert 'foo' in dest
+    assert 'bar/baz' not in dest
+    assert 'bar/qux' in dest
```

Comment:
Would be good to have some error tests? For example, what happens when a Group or Dataset/Array is already present? Do we have any incomplete copies for cases like this or is dest left unchanged?